### ISSUE TITLE
Read the probe scanners current progress.

### ISSUE THREAD

**Issue by Isfirs** [esi/esi-issues#930]
# Feature Request

It would be cool/handy to read the probe scanner data with the current scan progress.

## Use case

Makes wh mapping easier as it removes the need to copy&paste the probe scanner data.

## Authentication

remoteClientUI or a new scope `remoteProbeScanner`

## Example return

```json
[
  {
    "ID": "ABC-123",
    "type": "data|relic|gas|combat",
    "name": "Site name based on scan progress",
    "progress": 0.15
  }
]
```

# Checklist

Check all boxes that apply to this issue:

- [ ] Feature request description is provided
- [ ] Use case exists
- [ ] Feature requires a new route
- [ ] Feature adds data to existing route
- [ ] Feature requires new auth scope
- [ ] Feature can reuse existing scope
- [ ] Feature does not require auth
- [ ] Meta feature, applies to all routes

**Comment by lukasni** [comment]
this could probably go under the `/location/` route somehow, since all the `/ui/` routes are direct UI interactions.

**Comment by ficti0n1** [comment]
Now that scan results are persistent for the client between downtimes this could be expanded to show all probe progress/results for all sigs scanned. Would be brilliant for WH/sig mappers and would only include data that is already available to the user in game.

**Comment by ezi0o** [comment]
this would be amazing

**Comment by a-tal** [comment]
sent to GD

**Comment by Loroseco** [comment]
Good idea. You should only be able to read a sig when the type is known, to prevent any way to automatically detect new sigs spawning.

**Comment by skyride** [comment]
In addition to @andimiller point, it should also indicate the system_id these results are in reference to.

**Comment by cvweiss** [comment]
I think this would be an awesome feature and would remove many of the mundane tasks of working with wormhole mapping programs. Which may confuse some because of the conflict with my next statement:

I believe the users should have to do the work required in keeping their maps up to date and accurate. And this is coming from a guy who absolutely loves automation. If we provide an endpoint for signatures in the user's scanner, we might as well provide an endpoint for the user's local and the people in system. Some intel via api is just too much.

**Comment by Aidansavage** [comment]
In it's current "iteration" the endpoint would merely return nothing. Just as it would return nothing if the user doesnt do any scanning in a system with signatures.

That being said, I think it should be a "historical" sort of endpoint. Instead of giving real-time info, pending cache timers, it would return the information "stored" by the client, as the client has been updated by Psycho Sisters (I think that's the name of the team responsible?) to remember signature information, including scan %, between session changes that arent a result of logging off and logging back in. This would allow the endpoint to have a reduced chance of being used for intel if it required you to go back to a system and commit another scan to memory in order to inform the endpoint that a given sig has gone away.

WRT to having system information available in the return as well, I think that should hinge on whether or not this endpoint would require the location scope as well, due to this being able to find your location in some form.

**Comment by ccp-zoetrope** [comment]
From GD:
Approved for completed scans
Not approved for scans in progress.

**Comment by Loroseco** [comment]
Agreed, this endpoint needs to output partial scans to be at all useful. The best compromise would be to not output sigs at 0%

**Comment by skyride** [comment]
It really needs all sigs to work.

The usecase for this is simple, chainmapping tools.

Right now a fairly typical activity for a wormhole is:

1. Log in, open siggy, undock
2. Copy all sigs from the probe window into siggy. Old sigs that have
disapeared are removed. New sigs will show with no name.
3. Scan all new sigs and paste into siggy again thus populating their types.
4. Jump into connected holes and repeat this process.

Without 0% sigs, there is no way to distinguish between expired and
unscanned signatures.

**Comment by w9jds** [comment]
Personally, I think that if it isn't all of the signatures then just don't bother putting in the time to make the endpoint. My mapper accepts pastes from anywhere on screen and requires almost no work. If All signatures won't propagate from the API then the user still has to do the same thing. Even then I think this is pushing the boundaries of what should probably be pulled from the game through automation.

**Comment by Wizbit1977** [comment]
IF character is online and they have a full scanned signature list, i feel that the tools should reflect this, and only should show scanned signatures, be it partial or full, then they should be read into a mapper as scanned. i do not think there is any reason to show 0% signatures if you haven't entered the system to scan to get even partial signatures, why would the information even be relevant? therefore no untoward information is being presented from a character point and if it is, it is because they have started scanning and adding data to the mapper! it would help anyone scanning to have that feature and could only improve game play, without giving too much away

**Comment by NathenSample** [comment]
I'd agree with @w9jds; Without capturing of all sigs the user will still be left in a scenario where they're required to do manual input, or more realistically CTRL+A, CTRL+C, at which point the endpoints value is marginal; it limits the ability to report fake sigs, but ultimately you'd still need to trust user input.

Meaning the user still has to do manual input, and I still have to trust that input, meaning we there's not much of a net game.

On the other end of the spectrum if the data revealed was granular enough, i.e in that it updated all sigs simply by being present in the system, then we're definitely venturing deeply into the gray zone, yes that data is readily available in the client but do we want to automate the retrieval of it at that level? I don't think so.

I don't think GD are going to, nor should they, approve the exposure of all signatures, and I'm not sure the endpoint justifies the work without all sigs.

**Comment by webba** [comment]
I feel partial sigs would be nice for quality of life when scanning a chain initially I agree that without 0% manual input options would still be needed so the value added is marginal. I also don't feel that having chain mappers beep at you for new sigs in any system a member is in would be healthy, also any null group who utilizes their space and has every characters API could fairly quickly remove a lot of the work from exploration gameplay.

**Comment by obilque** [comment]
While this probably isn't the best place for it, I have open (and I assume there are many other bug reports) EBR-133297 about a year ago for signatures not being removed from the probe scanner when you are docked in a citadel OR not in system.  Status currently is "Attached" in my bug reports.

If this could be looked at long with these changes, that would be great because while this bug exists, signature probe results are somewhat unreliable because they are not properly reflective of the current state of the game under these circumstances.  It would be ideal for this endpoint to report the correct results, not the bugged version we have now.

**Comment by ezi0o** [comment]
**this endpoint needs to output partial scans to be at all useful. The best compromise would be to not output sigs at 0%**

**Comment by Isfirs** [comment]
I strongly disagree with that @ezi0o . To be consistent, this would require that you can't see 0% signatures in-game as well, which is not the case (except if that changed in the past 4 months that I didn't play). Also, in common sense of communication, refrain from using bold text if you want to put everything in bold anyway. Makes no sense at all.

**Comment by ezi0o** [comment]
> It really needs all sigs to work. The usecase for this is simple, chainmapping tools. Right now a fairly typical activity for a wormhole is: 1. Log in, open siggy, undock 2. Copy all sigs from the probe window into siggy. Old sigs that have disapeared are removed. New sigs will show with no name. 3. Scan all new sigs and paste into siggy again thus populating their types. 4. Jump into connected holes and repeat this process. Without 0% sigs, there is no way to distinguish between expired and unscanned signatures.
> […](#)

this man says it all

**Comment by Wizbit1977** [comment]
> > It really needs all sigs to work. The usecase for this is simple, chainmapping tools. Right now a fairly typical activity for a wormhole is: 1. Log in, open siggy, undock 2. Copy all sigs from the probe window into siggy. Old sigs that have disapeared are removed. New sigs will show with no name. 3. Scan all new sigs and paste into siggy again thus populating their types. 4. Jump into connected holes and repeat this process. Without 0% sigs, there is no way to distinguish between expired and unscanned signatures.
> > […](#)
> 
> this man says it all

i dont think you guys understand how the esi/api will work.. if you have partial or fully scanned sigs and they are linked to a jsig and all have their individual sig id's the mapper should autoupdate as you scan them, why would it not make sense to have the esi automatically recognise if you jumped into a system and then one of them all ready scanned sigs has gone? its not going to update unless you go through the systems 1 by 1 and update them.. difference is.. it will automagically update the mapper if sigid for that jsig has gone.. you still have to actively jump about.. however.. if you find a new sigid in your system then it will need to be partial or fully scanned to be auto accepted into the mapper.. also one thing the mapper would still require is wormhole types and such being set as per usual.. This does eliminate some of the tedium of checking sigs and such.. if it automagically updates the mapper if there is a dead sig on your list.. 0% just never need to be sent to esi

**Comment by NathenSample** [comment]
`why would it not make sense to have the esi automatically recognise if you jumped into a system and then one of them all ready scanned sigs has gone?`
Because a player who is not actively engaging in exploration content, should not simply by virtue of existing, be able to update exploration mappings.

`its not going to update unless you go through the systems 1 by 1 and update them.. difference is.. it will automagically update the mapper if sigid for that jsig has gone.. you still have to actively jump about.. however..`
Entities with thousands of people will be able to accurately map signature presence across the world, despite having no intent of performing exploration. The value of knowing a wormhole has collapsed is exactly that, value. And if you as a player want to bring value to the group/yourself with regards to that content, you should have to engage in it.

Also for the love of god, use punctuation and formatting.

**Comment by Wizbit1977** [comment]
_Because a player who is not actively engaging in exploration content, should not simply by virtue of existing, be able to update exploration mappings._

He isn't, he is only updating his personal list of sigs he has scanned,

_Entities with thousands of people will be able to accurately map signature presence across the world, despite having no intent of performing exploration. The value of knowing a wormhole has collapsed is exactly that, value. And if you as a player want to bring value to the group/yourself with regards to that content, you should have to engage in it._

Not true, only people who have an existing scanned list on the that character will be able to update a dead sig if it was already scanned.. therefore they would have had to engage in exploration to have an existing scanned sig on..

as for punctuation and formatting.. not really any of your business is it!!

**Comment by NathenSample** [comment]
You're expecting people to read and respond to you, common decency says you don't want to cause that person's eyes to bleed whilst doing so. Regardless, to the point at hand.

You say the person isn't updating maps - But he blatantly is. Consider the following: 

Person A goes through a system, and is a prober, he probes down a wormhole, your ratters safe up, because there's a wormhole nearby.
An hour later, person B goes by in a travel ceptor, he doesn't have probes, he isn't probing, he doesn't even consider the notion of looking at his scanner, but simply by virtue of existing, ESI is able to pull out the fact that the wormhole signature no longer exists in his system.

Subsequently, seeing that there is no longer a wormhole, your ratters undock again.

In this scenario, person B was able to provide up to date information via an automated system, despite him having no intent to do so, like I said, it's purely by virtue of him existing in that system. Yes, the system has have been mapped by a prober to know that signature was a wormhole, but the process of removing that should not be something automated.

**Comment by Wizbit1977** [comment]
_An hour later, person B goes by in a travel ceptor, he doesn't have probes, he isn't probing, he doesn't even consider the notion of looking at his scanner, but simply by virtue of existing, ESI is able to pull out the fact that the wormhole signature no longer exists in his system._

not sure if your being bloody minded or not.. but.. as per your example how the hell can person B update person A's sig list if he doesnt have a scanned sig on his character that now suddenly doesn't exist?

**Comment by NathenSample** [comment]
Your character, and subsequently the data it generates does not exist in a void. Think of it in terms of alliance tooling as opposed to an individual existing in a vacuum.

**Comment by Wizbit1977** [comment]
ok.. im not talking about alliance tooling.. i am referring to how the esi interprets information.. ideally only the sig list of person A will be updated by them.. it is attached to their esi info.. therefore only the updated info from Person A is sent to the tool..

**Comment by rayanth** [comment]
> ok.. im not talking about alliance tooling.. i am referring to how the esi interprets information.. ideally only the sig list of person A will be updated by them.. it is attached to their esi info.. therefore only the updated info from Person A is sent to the tool..
> 
> do you understand what i mean? because i still feel you don't!

There is nothing to stop the tool itself from sharing that information with other players. What @NathenSample is arguing, and I fully agree, is that an Alliance could include this in their alliance-management software of choice, have all of their members add this scope, and that management software would automatically update a mapper, visible only to the alliance, with all of the information every player is sharing from their own ESI endpoints to the management software. The players may not even realize they're doing it - they were told to provide this list of scopes, and they did.

With enough players flying about, a particularly large alliance could feasibly have a complete map of all wormholes, in a short amount of time. For that matter, a well-made public site could do the same. And then what's the point of exploration, if even the randomly-generated J-space is extremely well mapped, automatically? Where's the challenge and excitement?

I do think GD is correct in saying it shouldn't be all sigs, for exactly this reason. MAYBE saying scans at 75% or higher, so the full name is at least known and not just the type, but if you automate this too much, then the challenge and allure of J-space is gone, and all the best anomaly sites will be hunted over as soon as they've been spotted by anyone using the tool.
